code/modules/surgery/organs/organ_movement.dm

```diff
@@ -270,41 +283,41 @@
 	color = bodypart_overlay.draw_color //Defaults to the legacy behaviour of applying the color to the item.
 
 /// In space station videogame, nothing is sacred. If somehow an organ is removed unexpectedly, handle it properly
-/obj/item/organ/proc/forced_removal()
+/obj/item/organ/proc/forced_removal(datum/source, atom/old_loc, ...)
 	SIGNAL_HANDLER
 
 	if(owner)
-		Remove(owner)
+		if(loc?.loc == owner) // loc = some bodypart, loc.loc = some bodypart's owner
+			stack_trace("Forced removal triggered on a organ moving into the same mob!")
+		else
+			Remove(owner)
 	else if(bodypart_owner)
-		bodypart_remove(bodypart_owner)
+		if(loc == bodypart_owner)
+			stack_trace("Forced removal triggered on a organ moving into the same bodypart!")
```

Comment:
```suggestion
			stack_trace("Forced removal triggered on [src] moving into the same bodypart ([bodypart_owner])!")
```
Having specific details will come in handy if this triggers and (hopefully) reduce the guesswork